src/renderer/src/routes/modals/CreateWallet/CreateWalletStep4Modal.tsx

```diff
@@ -1,43 +1,118 @@
+    const wallet = createWallet({
+      name: nameTrimmed,
+      mnemonic: words.join(' '),
+    })
 
-  const handleNavigate = () => {
-    modalNavigate(-4)
-    navigate(`/app/wallets/${account.id}/overview`)
+    const selectedBlockchains = actionData.selectedBlockchains.filter(service => service.checked)
+
+    const accounts = await Promise.all(
+      selectedBlockchains.map(
+        async newAcc =>
+          await createStandardAccount({
+            wallet,
+            blockchain: newAcc.service,
+            name: commonT('account.defaultName', { accountNumber: 1 }),
+          })
+      )
+    )
+
+    modalNavigate('create-wallet-step-5', { state: { accounts } })
+  }
+
+  const handleSelectedBlockchain = (position: number) => {
+    setData({
+      selectedBlockchains: actionData.selectedBlockchains.map((service, index) => {
+        return index === position ? { ...service, checked: !service.checked } : { ...service }
+      }),
+    })
   }
 
   return (
     <CreateWalletModalLayout>
-      <div className="flex flex-col items-center w-full h-full justify-between">
-        <div className="flex flex-col w-full gap-2.5 px-28 items-center justify-center h-full">
-          <NeonWalletIcon3D />
-          <div className="text-white text-lg text-center">{t('title')}</div>
-          <div className="text-gray-100 text-xs text-center">{t('description')}</div>
+      <header className="flex justify-between items-center py-2.5">
+        <div className="flex items-center gap-x-2.5">
+          <MdLooks4 className="text-blue h-4.5 w-4.5" />
+          <h2 className="text-sm">{t('title')}</h2>
+        </div>
+        <div className="text-blue text-sm">{t('step4of4')}</div>
+      </header>
+      <Separator className="min-h-[0.0625rem] mb-9" />
+      <form
+        onSubmit={handleAct(handleSubmit)}
+        className="flex flex-col items-center w-full flex-grow justify-between min-h-0"
+      >
+        <div className="flex flex-col w-full gap-8 min-h-0">
+          <div className="text-gray-100 text-xs">{t('description')}</div>
+          <Separator />
+
+          <ul className="m-auto w-1/2 flex flex-col flex-grow gap-2 overflow-auto mb-4">
+            {blockchainNames.map((service, index) => (
+              <li key={index} className="flex h-12 rounded bg-asphalt border-none px-6 py-4">
+                <div className="flex justify-between flex-grow items-center">
+                  <label className="flex items-center gap-2.5" htmlFor={service}>
+                    <BlockchainIcon blockchain={service} type="gray" />
+                    {blockchainT(service)}
+                  </label>
+
+                  <Checkbox
+                    id={service}
+                    value={service}
+                    onCheckedChange={() => handleSelectedBlockchain(index)}
+                    checked={actionData.selectedBlockchains[index].checked}
+                    className="rounded"
+                  />
```

Comment:
The `<Checkbox />` should be inside of the `<label />` to improve the accessibility with click. So you won't need the generic `id` props more.